tensorflow_datasets/text/disco_fuse.py

```diff
@@ -0,0 +1,165 @@
+method on two corpora - sports articles crawled from the Web, and Wikipedia."""
+
+_DATA_OPTIONS = ['wikipedia', 'sports']
+_URLS = {'wikipedia': 'https://storage.googleapis.com/discofuse_dataset_v1/discofuse_v1_wikipedia.tar.gz',
+         'sports': 'https://storage.googleapis.com/discofuse_dataset_v1/discofuse_v1_sports.tar.gz'}
+
+
+class DiscoFuseConfig(tfds.core.BuilderConfig):
+  """BuilderConfig for DiscoFuse."""
+
+  @api_utils.disallow_positional_args
+  def __init__(self, **kwargs):
+    """BuilderConfig for DiscoFuse.
+    Args:
+      **kwargs: keyword arguments forwarded to super.
+    """
+    super(DiscoFuseConfig, self).__init__(**kwargs)
+
+
+class DiscoFuse(tfds.core.GeneratorBasedBuilder):
+  """DiscoFuse: A Large-Scale Dataset for Discourse-Based Sentence Fusion"""
+
+  VERSION = tfds.core.Version('0.1.0')
+  BUILDER_CONFIGS = [
+      DiscoFuseConfig(
+          name="wikipedia",
+          version="0.0.1",
+          description="DiscoFuse from Wikipedia Articles"),
+      DiscoFuseConfig(
+          name="sports",
+          version="0.0.1",
+          description="Sports from Sports Articles")
+  ]
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            "coherent_first_sentence": tfds.features.Text(),
+            "coherent_second_sentence": tfds.features.Text(),
+            "incoherent_first_sentence": tfds.features.Text(),
+            "incoherent_second_sentence": tfds.features.Text(),
+            "discourse_type": tfds.features.ClassLabel(names=[
+                'PAIR_ANAPHORA',
+                'PAIR_CONN',
+                'PAIR_CONN_ANAPHORA',
+                'PAIR_NONE',
+                'SINGLE_APPOSITION',
+                'SINGLE_CATAPHORA',
+                'SINGLE_CONN_INNER',
+                'SINGLE_CONN_INNER_ANAPHORA',
+                'SINGLE_CONN_START',
+                'SINGLE_RELATIVE',
+                'SINGLE_S_COORD',
+                'SINGLE_S_COORD_ANAPHORA',
+                'SINGLE_VP_COORD']),
+            "connective_string": tfds.features.Text(),
+            "has_coref_type_pronoun": tfds.features.ClassLabel(
+                num_classes=2),
+            "has_coref_type_nominal": tfds.features.ClassLabel(
+                num_classes=2)
+        }),
+        supervised_keys=None,
+        urls=['https://github.com/google-research-datasets/discofuse'],
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Returns SplitGenerators."""
+    url = _URLS[self.builder_config.name]
+    directory = dl_manager.download_and_extract(url)
+    directory = os.path.join(directory, self.builder_config.name)
+    train = os.path.join(directory, "train.tsv")
+    dev = os.path.join(directory, "dev.tsv")
+    test = os.path.join(directory, "test.tsv")
+    train_balanced = os.path.join(directory, "train.tsv")
```

Comment:
What are those `*_balanced` correspond too ? It seems they are using the same `train.tsv` data as the `train` split.

Reply:
I think this is a copy paste mistake, @sklan please correct it.

Reply:
Done.